src/api/app/services/gitlab_status_reporter.rb

```diff
@@ -0,0 +1,28 @@
+class GitlabStatusReporter < SCMStatusReporter
+  def call
+    gitlab_client = Gitlab.client(endpoint: "#{@event_subscription_payload[:api_endpoint]}/api/v4",
+                                  private_token: @scm_token)
+    # https://docs.gitlab.com/ce/api/commits.html#post-the-build-status-to-a-commit
+    gitlab_client.update_commit_status(@event_subscription_payload[:project_id],
+                                       @event_subscription_payload[:commit_sha],
+                                       @state,
+                                       status_options)
```

Comment:
After this line, we also need to add something like  `@workflow_run.save_scm_report_success(request_context) if @workflow_run.present?`

But let's wait until PR #12799  is merged because it changes that code a bit.